### PR TITLE
EIP-39 Monotonic box creation height

### PR DESCRIPTION
**Pull request by MrStahlfelge**

### CODE REVIEW

File: eip-0039.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+be done for every single contract on its own. That's why we suggest to add a check on protocol level.
+
+## Suggestion
+
+Suggestion to solve this problem on protocol level is to introduce a monotonic creation height rule as follows:
+
+For every transaction that is submitted to the network, it should be validated if the following expression is satisfied:
+
+    val highestInputBoxHeight = INPUTS.maxOf { input: Box => input.creationHeight }
+    OUTPUTS.allOf { output: Box => output.creationHeiht >= highestInputBoxHeight }
```
Comment (greenhat) [inline eip-0039.md]:
```suggestion
    OUTPUTS.allOf { output: Box => output.creationHeight >= highestInputBoxHeight }
```

**Review by greenhat — APPROVED** [review]

File: eip-0039.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# EIP-39 Monotonic box creation height rule
+
+* Author: MrStahlfelge
+* Status: Proposed
+* Created: 05-Oct-2022
+* Implemented: no
```
Comment (kushti) [inline eip-0039.md]: Now implemented

File: eip-0039.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Last edited: 05-Oct-2022
+* License: CC0
+* Forking: not needed for current draft
```
Comment (kushti) [inline eip-0039.md]: Soft-fork actually (new nodes will reject transactions breaking monotonic height rule, old nodes will accept both kinds of transactions).

**Review by kushti — CHANGES_REQUESTED** [review]
Please update index file also

**Review by kushti — APPROVED** [review]

**Comment by kushti** [comment]
@MrStahlfelge approved, but please update EIPs index (readme.md) before merging

**Pull request merged**